lib/backend/syncersv1/updateprocessors/rules_test.go

```diff
@@ -296,7 +299,119 @@ var _ = Describe("Test the Rules Conversion Functions", func() {
 		})
 
 		By("generating the correct destination selector", func() {
-			Expect(rulev1.SrcSelector).To(Equal(e))
+			Expect(rulev1.DstSelector).To(Equal(e))
+		})
+	})
+
+	It("should ignore the serviceaccount match", func() {
+
+		srce := fmt.Sprintf("(%s == 'namespace') && (has(label1))", apiv3.LabelNamespace)
+		r := apiv3.Rule{
+			Action: apiv3.Allow,
+			Source: apiv3.EntityRule{
+				ServiceAccounts: &apiv3.ServiceAccountMatch{Names: []string{"sa1", "sa2"},
+					Selector: "key == 'value1'",
+				},
+				Selector: "has(label1)",
+			},
+		}
+
+		// Process the rule and get the corresponding v1 representation.
+		rulev1 := updateprocessors.RuleAPIV2ToBackend(r, "namespace")
+
+		By("generating the correct source selector", func() {
+			Expect(rulev1.SrcSelector).To(Equal(srce))
+		})
+	})
+
+	It("should parse a serviceaccount match", func() {
+		os.Setenv("ALPHA_FEATURES", "serviceaccounts")
+		srce := fmt.Sprintf("(%s == 'namespace') && ((%skey == \"value1\") && (%s in {\"%s\", \"%s\"}))", apiv3.LabelNamespace, conversion.ServiceAccountLabelPrefix, apiv3.LabelServiceAccount, "sa1", "sa2")
+		dste := fmt.Sprintf("(pcns.nskey == \"nsvalue\") && ((%skey == \"value2\") && (%s in {\"%s\"}))", conversion.ServiceAccountLabelPrefix, apiv3.LabelServiceAccount, "sa3")
+
+		r := apiv3.Rule{
+			Action: apiv3.Allow,
+			Source: apiv3.EntityRule{
+				ServiceAccounts: &apiv3.ServiceAccountMatch{Names: []string{"sa1", "sa2"},
+					Selector: "key == 'value1'",
+				},
+			},
+			Destination: apiv3.EntityRule{
+				NamespaceSelector: "nskey == 'nsvalue'",
+				ServiceAccounts: &apiv3.ServiceAccountMatch{Names: []string{"sa3"},
+					Selector: "key == 'value2'",
+				},
+			},
+		}
+
+		// Process the rule and get the corresponding v1 representation.
+		rulev1 := updateprocessors.RuleAPIV2ToBackend(r, "namespace")
+
+		By("generating the correct source selector", func() {
+			Expect(rulev1.SrcSelector).To(Equal(srce))
+		})
+
+		By("generating the correct destination selector", func() {
+			Expect(rulev1.DstSelector).To(Equal(dste))
+		})
+	})
+
+	It("should parse a serviceaccount match with default namespace and no namespace selector", func() {
```

Comment:
this should say "global" rather than default namespace - passing `""` to `RuleAPIV2ToBackend` means a GNP, not the default namespace.

For the default ns, it will get "default" passed in.